### PR TITLE
Minor code cleanups around invoicing assignment

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor code cleanup

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------
This is follow up cleanup on the extraction

The biggest change is moving the assignments back up close to the main tplLineItems block but I'm pretty sure that's not going to be a problem

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire were you going to take a look at this

File: CRM/Contribute/Form/Contribution/Confirm.php [hunk 1]
```diff
@@ -508,18 +508,22 @@ public function buildQuickForm() {
...
       $this->assign('totalTaxAmount', $params['tax_amount']);
     }
+    if ($this->_priceSetId && !CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', $this->_priceSetId, 'is_quick_config')) {
+      $this->assign('lineItem', $tplLineItems);
+    }
+    else {
+      $this->assign('is_quick_config', 1);
+      $this->_params['is_quick_config'] = 1;
+    }
+
```
Comment (mattwire) [inline CRM/Contribute/Form/Contribution/Confirm.php]: No problem with moving this block, but can't see an obvious reason to do so?  Is this supporting further refactoring?

File: CRM/Financial/Form/FrontEndPaymentFormTrait.php [hunk 2]
```diff
@@ -50,12 +50,10 @@ protected function alterLineItemsForTemplate($tplLineItems) {
     $getTaxDetails = FALSE;
     foreach ($tplLineItems as $key => $value) {
       foreach ($value as $k => $v) {
-        if (isset($v['tax_rate'])) {
-          if ($v['tax_rate'] != '') {
-            $getTaxDetails = TRUE;
-            // Cast to float to display without trailing zero decimals
-            $tplLineItems[$key][$k]['tax_rate'] = (float) $v['tax_rate'];
-          }
+        if (isset($v['tax_rate']) && $v['tax_rate'] != '') {
```
Comment (mattwire) [inline CRM/Financial/Form/FrontEndPaymentFormTrait.php]: This is fine.  I fell into a trap which @seamuslee001 pointed out that you could have a 0% tax rate specifically defined which is why we don't want to collapse to empty.  Maybe worth a comment (for me next time I'm here) `// Don't use empty as a 0% tax rate may be ok.`

**Review by mattwire — CHANGES_REQUESTED** [review]
@eileenmcnaughton One question, one request to add a comment.  I've worked through this and have tested.

File: CRM/Contribute/Form/Contribution/Confirm.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Contribute/Form/Contribution/Confirm.php]: yep - pretty much

**Pull request merged**